File: Arrays/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numToIndex = new HashMap<>();
        int[] result = new int[2];

        for (int i = 0; i < nums.length; ++i) {
            int complement = target - nums[i];
            if (numToIndex.containsKey(complement)) {
                result[0] = numToIndex.get(complement);
                result[1] = i;
                break;
            }
            numToIndex.put(nums[i], i);
        }
        return result;
    }
}

File: Arrays/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    std::unordered_map<int, int> numToIndex;
    std::vector<int> result;

    for (int i = 0; i < nums.size(); ++i) {
        int complement = target - nums[i];
        if (numToIndex.find(complement) != numToIndex.end()) {
            result.push_back(numToIndex[complement]);
            result.push_back(i);
            break;
        }
        numToIndex[nums[i]] = i;
    }
        return result;
    }
};